pkg/cmd/admin/groups/changemembership.go

```diff
@@ -0,0 +1,142 @@
+package groups
+
+import (
+	"errors"
+	"fmt"
+	"io"
+
+	kcmdutil "github.com/GoogleCloudPlatform/kubernetes/pkg/kubectl/cmd/util"
+	"github.com/GoogleCloudPlatform/kubernetes/pkg/util"
+
+	"github.com/spf13/cobra"
+
+	"github.com/openshift/origin/pkg/client"
+	"github.com/openshift/origin/pkg/cmd/util/clientcmd"
+)
+
+const (
+	AddRecommendedName = "add-users"
+	addLong            = `
+Add users to a group.
+
+This command will append unique users to the list of members for a group.`
+
+	addExample = `  // Add user1 and user2 to my-group
+  $ %[1]s my-group user1 user2`
+)
+
+const (
+	RemoveRecommendedName = "remove-users"
+	removeLong            = `
+Remove users from a group.
+
+This command will remove users from the list of members for a group.`
+
+	removeExample = `  // Remove user1 and user2 from my-group
+  $ %[1]s my-group user1 user2`
+)
+
+type GroupModificationOptions struct {
+	GroupClient client.GroupInterface
+
+	Group string
+	Users []string
+}
+
+func NewCmdAddUsers(name, fullName string, f *clientcmd.Factory, out io.Writer) *cobra.Command {
+	options := &GroupModificationOptions{}
+
+	cmd := &cobra.Command{
+		Use:     name + " GROUP USER [USER ...]",
+		Short:   "Add users to a group",
+		Long:    addLong,
+		Example: fmt.Sprintf(addExample, fullName),
+		Run: func(cmd *cobra.Command, args []string) {
+			if err := options.Complete(f, args); err != nil {
+				kcmdutil.CheckErr(kcmdutil.UsageError(cmd, err.Error()))
+			}
+
+			kcmdutil.CheckErr(options.AddUsers())
+		},
+	}
+
+	return cmd
+}
+
+func NewCmdRemoveUsers(name, fullName string, f *clientcmd.Factory, out io.Writer) *cobra.Command {
+	options := &GroupModificationOptions{}
+
+	cmd := &cobra.Command{
+		Use:     name + " GROUP USER [USER ...]",
+		Short:   "Remove users from a group",
+		Long:    removeLong,
+		Example: fmt.Sprintf(addExample, removeExample),
+		Run: func(cmd *cobra.Command, args []string) {
+			if err := options.Complete(f, args); err != nil {
+				kcmdutil.CheckErr(kcmdutil.UsageError(cmd, err.Error()))
+			}
+
+			kcmdutil.CheckErr(options.RemoveUsers())
+		},
+	}
+
+	return cmd
+}
+
+func (o *GroupModificationOptions) Complete(f *clientcmd.Factory, args []string) error {
+	if len(args) < 2 {
+		return errors.New("You must specify at least two arguments: GROUP USER [USER ...]")
+	}
+
+	o.Group = args[0]
+	o.Users = append(o.Users, args[1:]...)
+
+	osClient, _, err := f.Clients()
+	if err != nil {
+		return err
+	}
+
+	o.GroupClient = osClient.Groups()
+
+	return nil
+}
+
+func (o *GroupModificationOptions) AddUsers() error {
+	group, err := o.GroupClient.Get(o.Group)
+	if err != nil {
+		return err
+	}
+
+	existingUsers := util.NewStringSet(group.Users...)
+	for _, user := range o.Users {
+		if existingUsers.Has(user) {
```

Comment:
> do we not dedup server-side?

No.  You gave us a group, it should be the group you mean.  I was disinclined to change their input simply because it could be simplified.